converters/fromZigbee.js

```diff
@@ -1581,7 +1581,11 @@ const converters = {
         cluster: 'genOnOff',
         type: ['attributeReport', 'readResponse'],
         convert: (model, msg, publish, options, meta) => {
-            return {click: getKey(model.endpoint(msg.device), msg.endpoint.ID)};
```

Comment:
What I mean is to change this converter to e.g.

```js
WXKG02LM_click: {
    cluster: 'genOnOff',
    type: ['attributeReport', 'readResponse'],
    convert: (model, msg, publish, options, meta) => {
        return {action: getProperty('single', msg, model)};
    },
},
legacy_WXKG02LM_click: {
    legacy: true
    cluster: 'genOnOff',
    type: ['attributeReport', 'readResponse'],
    convert: (model, msg, publish, options, meta) => {
        return {click: getKey(model.endpoint(msg.device), msg.endpoint.ID)};
    },
},
```

and change the definition to 

```js
{
    zigbeeModel: ['lumi.sensor_86sw2', 'lumi.sensor_86sw2.es1', 'lumi.remote.b286acn01'],
    model: 'WXKG02LM',
    vendor: 'Xiaomi',
    description: 'Aqara double key wireless wall switch',
    supports: 'left, right, both click (and double, long click for left, right and both depending on model)',
    fromZigbee: [
        fz.xiaomi_battery_3v, fz.WXKG02LM_click, fz.legacy_WXKG02LM_click,
        fz.WXKG02LM_click_multistate,
    ],
    toZigbee: [],
    meta: {multiEndpoint: true},
    endpoint: (device) => {
        return {'left': 1, 'right': 2, 'both': 3};
    },
    onEvent: xiaomi.prevent_reset,
},
```

Based on the `legacy` attribute we can automatically disable the old converted, note that I prefer `legacy` over `legacy_click` because not only the `click` can become legacy but also other stuff.

Also note that I use `getProperty`, this will make the `action` in the form of `action_endpoint` instead of just `endpoint`.

Reply:
Just out of curiosity  as I am not really familiar with the project: How is the correct converter selected If you provide a legacy and non legacy one? 
My understanding was that you can just set **all** `legacy` elements On or Off -> then It would be helpful to allow users to only enable a part of the legacy code. But it seems that it is individually decided per device?

Reply:
Ignore my comment about adding `legacy: true` property for the legacy converters. Changing the code from e.g.

```js
legacy_genOnOff_cmdOn: {
    cluster: 'genOnOff',
    type: 'commandOn',
    convert: (model, msg, publish, options, meta) => {
        return {click: 'on'};
    },
},
```

to 

```js
legacy_genOnOff_cmdOn: {
    cluster: 'genOnOff',
    type: 'commandOn',
    convert: (model, msg, publish, options, meta) => {
        if (options.hasOwnProperty('legacy') && options.legacy === false) {
            return {click: 'on'};
        }
    },
},
```

will allow to disable the legacy behaviour when setting the following in the Zigbee2MQTT `configuration.yaml`.

```yaml
devices:
  '0x12345678':
    friendly_name: my_device
    legacy: false
```

Reply:
I have submitted an update with the condition in each legacy function.